webrtc/README.md

```diff
@@ -0,0 +1,445 @@
+
+### `RTCDataChannel` label
+
+`RTCPeerConnection.createDataChannel()` requires passing a `label` for the
+to-be-created `RTCDataChannel`. When calling `createDataChannel` implementations
+MUST pass an empty string. When receiving an `RTCDataChannel` via
+`RTCPeerConnection.ondatachannel` implementations MUST NOT require `label` to be
+an empty string. This allows future versions of this specification to make use
+of the `RTCDataChannel` `label` property.
+
+## Connection Security
+
+Note that the below uses the message framing described in
+[multiplexing](#multiplexing).
+
+While WebRTC offers confidentiality and integrity via TLS, one still needs to
+authenticate the remote peer by its libp2p identity.
+
+After [Connection Establishment](#connection-establishment):
+
+1. _A_ and _B_ open a WebRTC data channel with `id: 0` and `negotiated: true`
+   ([`pc.createDataChannel("", {negotiated: true, id:
+   0});`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/createDataChannel)).
+
+2. _B_ starts a Noise `XX` handshake on the new channel. See
+   [noise-libp2p](https://github.com/libp2p/specs/tree/master/noise).
+
+   _A_ and _B_ use the [Noise
+   Prologue](https://noiseprotocol.org/noise.html#prologue) mechanism. More
+   specifically _A_ and _B_ set the Noise _Prologue_ to
+   `libp2p-webrtc-noise:<FINGERPRINT_A><FINGERPRINT_B>` before starting the
+   actual Noise handshake. `<FINGERPRINT_A><FINGERPRINT_B>` is the concatenation
+   of the two TLS fingerprints of _A_ (Noise handshake responder) and then _B_
+   (Noise handshake initiator), in their multihash byte representation.
+
+   On Chrome _A_ can access its TLS certificate fingerprint directly via
+   `RTCCertificate#getFingerprints`. Firefox does not allow _A_ to do so. Browser
+   compatibility can be found
+   [here](https://developer.mozilla.org/en-US/docs/Web/API/RTCCertificate). In
+   practice, this is not an issue since the fingerprint is embedded in the local
+   SDP string.
+
+3. On success of the authentication handshake, the used datachannel is
+   closed and the plain WebRTC connection is used with its multiplexing
+   capabilities via datachannels. See [Multiplexing](#multiplexing).
+
+Note: WebRTC supports different hash functions to hash the TLS certificate (see
+https://datatracker.ietf.org/doc/html/rfc8122#section-5). The hash function used
+in WebRTC and the hash function used in the multiaddr `/certhash` component MUST
+be the same. On mismatch the final Noise handshake MUST fail.
+
+_A_ knows _B_'s fingerprint hash algorithm through _B_'s multiaddr. _B_ could
+infer _A_'s fingerprint hash algorithm through _A_'s TLS certificate signature.
+Or _B_ could assume _A_ to use the same hash algorithm it discovers through
+_B_'s multiaddr. For now implementations MAY assume both _A_ and _B_ to use
+sha-256. Future iterations of this specification may add support for other hash
+algorithms in a backwards compatible way.
```

Comment:
To be future-proof, it would be nice to always use the hash function from the multiaddr. That way, when / if sha256 is broken, no implementation changes will be needed.

Reply:
Sounds good to me. Done with 602f492.

@melekes  @ckousik @john-LittleBearLabs if I am not mistaken rust-, js- and go-libp2p assume sha-256 everywhere. With the above change in the specification, implementations just need to discard multiaddrs that don't use sha-256 when doing outgoing connections.